v1/ast/term.go

```diff
@@ -2540,6 +2531,9 @@ func (obj *object) get(k *Term) *objectElem {
 	case Number:
 		if xi, ok := x.Int64(); ok {
 			equal = func(y Value) bool {
+				if x == y {
```

Comment:
Since `x` and `y` are interface types, is this just doing a pointer comparison, or is go smarter than so?